src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -200,22 +200,36 @@ public class ColumnFamilyStore implements ColumnFamilyStoreMBean, Memtable.Owner
     are finished. By having flushExecutor size the same size as each of the perDiskflushExecutors we make sure we can
     have that many flushes going at the same time.
     */
-    private static final ExecutorPlus flushExecutor = executorFactory()
+    private static ExecutorPlus flushExecutor;
+
+    // post-flush executor is single threaded to provide guarantee that any flush Future on a CF will never return until prior flushes have completed
+    private static ExecutorPlus postFlushExecutor;
+
+    private static ExecutorPlus reclaimExecutor;
+
+    private static PerDiskFlushExecutors perDiskflushExecutors;
+
+    static
+    {
+        if (!DatabaseDescriptor.isClientOrToolInitialized())
```

Comment:
this needs to be completely skipped in the client mode because DatabaseDescriptor is not initialized yet so it would fail to instantiate these executors because invalid values would be passed to that. Interestingly, the fact these fields are left to be null does not hurt anything because they are never accessed.

Reply:
I kept these final in https://github.com/apache/cassandra/commit/fecac728ec945e9b376a5fb2c4af54f337d744cf. Even if you don't take that patch wholesale, we should keep that part.